### PR TITLE
Add flang

### PR DESCRIPTION
**Pull request by isuruf**
cc @xoviat

### CODE REVIEW

File: recipes/flang/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  version: {{ version }}
+
+source:
+  url: https://github.com/isuruf/flang/archive/{{ flang_commit }}.tar.gz
+  sha256: {{ sha256 }}
+
+build:
+  number: {{ build_number }}
+  skip: true    # [osx]
+  skip: true    # [win32]
```
Comment (xoviat) [inline recipes/flang/meta.yaml]: You need to skip Python 2.7. There's no point in adding support because the ABI is incompatible.

Comment (xoviat) [reply recipes/flang/meta.yaml]: Also, why is win32 skipped?

Comment (isuruf) [reply recipes/flang/meta.yaml]: Flang only supports 64-bit right?

Comment (isuruf) [reply recipes/flang/meta.yaml]: There's no python in the build requirements. just `msvc_compiler: 14.0`.

Comment (xoviat) [reply recipes/flang/meta.yaml]: In theory yes (although I would like to understand why this is in the future). However, to me "win32" means both 32 and 64 bit windows. Maybe it's different for conda.

Comment (xoviat) [reply recipes/flang/meta.yaml]: So the test failures are okay then?

**Comment by xoviat** [comment]
@isuruf I think this is ready to go. Conda doesn't *have to* supply a linux flang initially because linux has gfortran, which is ABI compatible, while windows doesn't. We can always iterate more.

**Comment by xoviat** [comment]
Wait, I take that back.

**Comment by xoviat** [comment]
ref conda-forge/clangdev-feedstock#27

**Comment by isuruf** [comment]
@conda-forge/staged-recipes, this is ready now.

**Comment by scopatz** [comment]
Looks like circle is failing

**Comment by xoviat** [comment]
Not relevant for now because the primary benefit is for windows.

**Comment by xoviat** [comment]
Though the appveyor build should pass.

**Comment by jakirkham** [comment]
Should there be a skip for Linux then?

**Comment by xoviat** [comment]
Flang theoretically builds on linux; that problem should be resolved. However, *everyone* is waiting for a windows fortran compiler, so I would recommend resolving the problems with linux later.

**Comment by isuruf** [comment]
Yes, I'll skip Linux for now

**Comment by xoviat** [comment]
Looks like windows passed.

**Comment by isuruf** [comment]
Linux tests pass now. I think the earlier Linux failure was because of a build system issue with parallel building.

**Comment by xoviat** [comment]
I suspected nondeterminism in the linux builds. However, I would *really* not prefer debugging that at this point.

**Comment by xoviat** [comment]
@isuruf Ready.

**Comment by isuruf** [comment]
I'll let @jakirkham, @scopatz review this one.

**Comment by isuruf** [comment]
Ignore the travis-ci build. OSX is skipped.

**Comment by xoviat** [comment]
Any eta?

**Comment by scopatz** [comment]
You are right, this looks good to me then!

**Pull request merged**